app/assets/javascripts/rentals/new.js.erb

```diff
@@ -48,17 +47,65 @@ function costCalculation(){
         amount.val(data);
         date_range.val(range);
       },
-      error: function () {
-        alert('failed to retreive rental cost');
-      }
+      error: function () { alert('failed to retreive rental cost'); }
     });
    }
 }
 
+
+function userResultHandleClick(e) {
+  var userId = $(this).attr("id").substr(10); //select the user id
+  $("#rental_user_id").val(userId);
+
+  var row = $(this).parent().parent(); //select the row
+  var name = row.find("td:nth-child(1)").text() + " " + row.find("td:nth-child(2)").text(); //first two td's joined by space
+  $("#rental_userSearchQuery").val(name); //set the user to the full name
+
+  $("#modalSearchResults").modal('toggle'); //close modal
+}
+
+function ajaxUserSearch(pageNo) {
+  var query = $("#rental_userSearchQuery").val();
+
+  //inject page into request
+  var params = $.extend({user_search_query: query}, pageNo ? {page: pageNo} : {page: "1"} );
+
+  $.ajax({
+    url: Routes.rentals_search_users_path(),
+    dataType: "html",
+    data: params,
+    success: function (data) {
+      $("#userResults").html(data); //put content in the modal
+
+      //attach event listeners to results
+      $(".userResult").click(userResultHandleClick);
+
+      //need to reformat links
+      $("#userResults .pagination a").each(modalSearchResultsFormatPaginationLinks);
+    },
+    error: function () { alert("Failed to retreive users."); }
+  });
+}
+
+function getQueryString( field, url ) {
```

Comment:
In that case, it belongs in `vendor`, and its source should be credited.

But I think URI.js is nicer than this, and if you include it properly as a library, like with rails-assets, then you'll get updates over time as the maintainers of that library make patches and bugfixes.

Whereas in its current form, that load is on us.